### PR TITLE
[Not urgent!] LF-4384(b): Refactor useAnimalOrBatchRemoval

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

Currently the `useAnimalOrBatchRemoval` hook accepts inventory IDs and parses them. This parsing is redundant for the `SingleAnimalView`, which already has a parsed ID. This PR makes the hook more generic by:

- Updating `useAnimalOrBatchRemoval` to accept:
   - `id`s with `kind` (animal or batch key) instead of inventory IDs
   - `onSuccess` instead of `setSelectedInventoryIds`
- Move the inventory ID parsing and `setSelectedInventoryIds` logic to the `Inventory` component.

(This PR addresses the [comment](https://github.com/LiteFarmOrg/LiteFarm/pull/3534#discussion_r1856718064) on https://github.com/LiteFarmOrg/LiteFarm/pull/3534)

### CODE REVIEW

File: packages/webapp/src/containers/Animals/Inventory/index.tsx [hunk 1]
```diff
@@ -102,9 +105,18 @@ function AnimalInventory({
     [updateSelectedTypeIds],
   );
 
+  const onRemovalSuccess = (animalOrBatchKey: AnimalOrBatchKeys, ids: Animal['id'][]) => {
+    setSelectedInventoryIds((selectedInventoryIds) => {
+      return selectedInventoryIds.filter((i) => {
+        const { kind, id } = parseInventoryId(i);
+        return kind === animalOrBatchKey && ids.includes(id);
```
Comment (Duncan-Brain) [inline packages/webapp/src/containers/Animals/Inventory/index.tsx:112]: The previous filter was : `!selectedAnimalIds.includes(i))` 

Everything looks like it works so I am not sure it matters, but it looks like it is setting the selected inventory id to the ones that were successfully removed?

**Review by Duncan-Brain — DISMISSED** [review]
Left an inline comment but I am not sure it is valid.

File: packages/webapp/src/containers/Animals/Inventory/index.tsx (hunk 1 above)
Comment (SayakaOno) [reply packages/webapp/src/containers/Animals/Inventory/index.tsx]: Thank you so much for catching it! I remember getting confused about how to use `.filter()` suddenly and ended up flipping the condition! 🙏
It's fixed now!

**Review by Duncan-Brain — APPROVED** [review]
Still works for me !

**Comment by SayakaOno** [comment]
Converting to draft to finalize after https://github.com/LiteFarmOrg/LiteFarm/pull/3554 is merged. (Otherwise this will get conflicts again 😢)

**Comment by SayakaOno** [comment]
We can revisit this once we have time after the release, as more conflicts are expected.

**Pull request closed without merging**